cassdegrees/ui/views.py

```diff
@@ -95,8 +95,61 @@ def sampleform(request):
         return render(request, 'sampleform.html')
 
 
+# Using sampleform template and #59 - basic degree creation workflow as it's inspirations
 def create_subplan(request):
-    return render(request, 'createsubplan.html')
+
+    render_properties = {
+        'msg': None,
+        'is_error': False,
+        'code': None,
+        'year': None,
+        'name': None,
+        'planType': None
+    }
+
+    if request.method == 'POST':
+        model_api_url = 'http://127.0.0.1:8000/api/model/subplan/'
+        post_data = request.POST
+
+        # Generate units from subtype plan selected
+        subplanUnits = \
+            {
+                'MAJ': 48,
+                'MIN': 24,
+                'SPEC': 24
+            }
+
+        subplanfields = \
+            {
+                'code': post_data.get('code'),
+                'year': post_data.get('year'),
+                'name': post_data.get('name'),
+                'units': subplanUnits[post_data.get('planType')],
+                'planType': post_data.get('planType')
+            }
+
+        # Submit a POST request to the model with subplan data
+        rest_api = requests.post(model_api_url, data=subplanfields)
+
+        # Store fields in render properties so they can be repopulated on error or success
+        for field in subplanfields:
+            render_properties[field] = subplanfields[field]
+
+        # Handle request return type and generate success or fail message
+        if rest_api.status_code == 201:
+            render_properties['msg'] = 'Subplan successfully added.'
+        else:
+            render_properties['is_error'] = True
+
+            rest_response = rest_api.json()
+            if "The fields code, year must make a unique set." in rest_response['non_field_errors']:
+                render_properties['msg'] = "A subplan already exists with this code and year."
+            elif "The fields year, name must make a unique set." in rest_response['non_field_errors']:
+                render_properties['msg'] = "A subplan already exists with this name and year."
```

Comment:
I just realised that Majors and Minors can have the same name sometimes:
[ARAB-MAJ](https://programsandcourses.anu.edu.au/2019/major/ARAB-MAJ), [ARAB-MIN](https://programsandcourses.anu.edu.au/2019/minor/ARAB-MIN)

Might be a good idea to fix this. Either we should remove the `Name-Year` restriction or enforce a `Name-Year-Type` restriction

Reply:
Have changed the restriction to Name-Year-Type for the moment to resolve this issue, probably preferable to removing the restriction altogether, but happy to revisit.